src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,156 @@
+package com.fakng.fakngagrgtr.parser.stripe;
+
+import com.fakng.fakngagrgtr.parser.HtmlParser;
+import com.fakng.fakngagrgtr.parser.LocationProcessor;
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.DomElement;
+import com.gargoylesoftware.htmlunit.html.DomNode;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.*;
+
+@Component
+public class StripeParser extends HtmlParser {
+
+    private static final int pageSize = 100;
+    private static final String STRIPE_URL = "https://stripe.com";
+    private static final String JOBS_PAGINATION_LIST_XPATH = "//a[text()='Last']";
+    private static final String VACANCY_TITLE_XPATH = "//h1[@class='Copy__title']";
+    private static final String VACANCIES_URLS_XPATH = "//a[@data-analytics-label='jobs_listings_title_link']";
+    private static final String VACANCY_DESCRIPTION_BLOCK_XPATH = "//div[@class='ArticleMarkdown']//*";
+    private static final String VACANCY_CITIES_XPATH = "//span[@class='JobsListings__locationDisplayName']";
+    private static final String VACANCY_JOB_TYPE_XPATH = "//p[text()='Job type']";
+    private static final String VACANCY_TEAM_XPATH = "//p[text()='Team']";
+
+    public StripeParser(WebClient htmlWebClient,
+                        CompanyRepository companyRepository,
+                        LocationProcessor locationProcessor,
+                        @Value("${url.stripe}") String url) {
+        super(htmlWebClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Stripe";
+    }
+    @Override
+    public List<Vacancy> getAllVacancies() throws IOException {
+        HtmlPage page = downloadPage(String.format(url, 0));
+        int pageCount = getPageCount(page);
+        List<Vacancy> vacancyList = new ArrayList<>();
+
+        for (int i = 0; i <= pageCount; i++) {
+            page = downloadPage(String.format(url, i));
+            vacancyList.addAll(fillVacancy(
```

Comment:
Why do you need to pass those two params here, when, in fact, they both can be received from one - page?

Reply:
I need to get country code and the only way to get country code its from vacancy listing page

Reply:
I understand that. But do you need to store it in a separate object, or can just put it directly into Vacancy object after parsing? What the point of keeping VacancyDTO here?

Reply:
I understand that. But can't you retrieve those DomElements inside the method and pass only page as parameter? This code becomes ugly, if you pass those separate elements inside